lib/ramen/test.js

```diff
@@ -0,0 +1,152 @@
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+        analytics.initialize();
+        analytics.page();
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+      });
+
+      it('should call #load', function(){
+        analytics.initialize();
+        analytics.page();
+        analytics.called(ramen.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(ramen, done);
+    });
+  });
+
+  describe('after loading', function(){
+    beforeEach(function(done){
+      analytics.once('ready', done);
+      analytics.initialize();
+    });
+
+    describe('#identify', function(){
+      beforeEach(function(){
+        analytics.stub(window.Ramen, 'go');
+      });
+
+      it('should not call Ramen.go if only id is passed', function(){
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if just email passed', function(){
+        var email = 'email@example.com';
+        analytics.identify('id', { email: email });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === email);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if email & name passed', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        analytics.identify('id', { email: email, name: name });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === name);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should pass along integration options', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        var auth_hash = 'authy_hasy';
+        var timestamp = new Date() / 1000;
+        var custom_links = [{href: 'https://ramen.is/support', title: 'Hello'}];
+        var labels = ['use', 'ramen!'];
+        var environment = 'staging';
+        var logged_in_url = 'https://align.ramen.is/manage';
+        var unknown_future_opt = '11';
+        var unknown_future_user_opt = 'user 11';
+
+        analytics.identify('id', {email: email, name: name},
+          {
+            integrations: {
+              Ramen: {
```

Comment:
Sure, let me know if you'd like me to put these in the source code as comments:

* `unknown_future_opt` is simply to test that anything passed in will be added to the `window.ramenSettings` object
* `environment` is the application environment. eg. "production", "staging", "development", etc... 
* `auth_hash` is SHA256 of "[user email]:[user id]:[user name]:[timestamp]:[secret key]" -- If this is sent in and `environment` is production, then we will automatically enable "Secure Mode" and all future requests will require the `auth_hash`
* `custom_links` tells us what to display at the bottom of the Ramen "Product Center Tray". In [this image](https://dl.dropboxusercontent.com/u/2519742/ramen-custom-links.png) the lines of text at the bottom are set by `custom_links`
* `user.unknown_future_user_opt` is analogous to `unknown_future_opt` but for the `window.ramenSettings.user` object
* `labels` is an array of strings that can be applied to a `user` and used on our end for targeting. For example `["Gold", "Paid", "Used Coupon"]`
* `logged_in_url` tells us a URL we can send a customer to that will guarantee they are logged in on and are being exposed to the RamenJS. This is so that in an email (eg. a new comment notification) we can have a 'view discussion' link that will 1) send them to Ramen 2) and cookie them 3) before redirecting them to `logged_in_url` where 4) RamenJS will see the cookie and popup the appropriate discussion